opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -81,33 +88,32 @@ public class OpenSearchExprValueFactory {
 
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-  private final Map<ExprType, Function<Content, ExprValue>> typeActionMap =
-      new ImmutableMap.Builder<ExprType, Function<Content, ExprValue>>()
-          .put(INTEGER, c -> new ExprIntegerValue(c.intValue()))
-          .put(LONG, c -> new ExprLongValue(c.longValue()))
-          .put(SHORT, c -> new ExprShortValue(c.shortValue()))
-          .put(BYTE, c -> new ExprByteValue(c.byteValue()))
-          .put(FLOAT, c -> new ExprFloatValue(c.floatValue()))
-          .put(DOUBLE, c -> new ExprDoubleValue(c.doubleValue()))
-          .put(STRING, c -> new ExprStringValue(c.stringValue()))
-          .put(BOOLEAN, c -> ExprBooleanValue.of(c.booleanValue()))
+  private final Map<ExprType, BiFunction<Content, MappingEntry, ExprValue>> typeActionMap =
+      new ImmutableMap.Builder<ExprType, BiFunction<Content, MappingEntry, ExprValue>>()
```

Comment:
Because `MappingEntry` is only used by date time, I'm think can we add this mapping info to `Content`?